test/edu/washington/escience/myria/operator/CatalogScanTest.java

```diff
@@ -0,0 +1,75 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.operator;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.api.encoding.QueryEncoding;
+import edu.washington.escience.myria.coordinator.catalog.CatalogException;
+import edu.washington.escience.myria.coordinator.catalog.MasterCatalog;
+import edu.washington.escience.myria.storage.TupleBatch;
+
+/**
+ * 
+ */
+public class CatalogScanTest {
+
+  /**
+   * The catalog
+   */
+  private MasterCatalog catalog;
+
+  /**
+   * @throws java.lang.Exception
+   */
+  @Before
+  public void setUp() throws Exception {
+    /* Turn off SQLite logging, it's annoying. */
+    Logger.getLogger("com.almworks.sqlite4java").setLevel(Level.OFF);
+
+    catalog = MasterCatalog.createInMemory();
+
+    // add a query to the catalog
+    QueryEncoding query = new QueryEncoding();
+    query.rawQuery = "query 1 is about baseball";
+    query.logicalRa = "";
+    catalog.newQuery(query);
+  }
+
+  /**
+   * Destroy catalog.
+   */
+  @After
+  public void Cleanup() {
+    catalog.close();
+  }
+
+  @Test
+  public final void testQueryQueries() throws DbException, CatalogException {
+    Schema schema = new Schema(ImmutableList.of(Type.LONG_TYPE, Type.STRING_TYPE), ImmutableList.of("id", "raw"));
```

Comment:
`Schema.ofFields()`, or at least `Schema.of()` (we need to do something about public exposure of the `Schema` constructor).